server/wfprev-api/src/main/java/ca/bc/gov/nrs/wfprev/services/ActivityService.java

```diff
@@ -0,0 +1,233 @@
+import ca.bc.gov.nrs.wfprev.data.repositories.ProjectFiscalRepository;
+import ca.bc.gov.nrs.wfprev.data.repositories.RiskRatingCodeRepository;
+import jakarta.persistence.EntityNotFoundException;
+import jakarta.transaction.Transactional;
+import jakarta.validation.ConstraintViolationException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.hateoas.CollectionModel;
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+@Slf4j
+@Component
+public class ActivityService implements CommonService {
+
+    private static final String FISCAL_NOT_FOUND = "Project Fiscal not found: ";
+    private static final String PROJECT_FISCAL = "Project Fiscal ";
+    private static final String DOES_NOT_BELONG_PROJECT = " does not belong to Project ";
+    private static final String DOES_NOT_BELONG_FISCAL = " does not belong to Project Fiscal ";
+    private static final String ACTIVITY_NOT_FOUND = "Activity not found: ";
+    private static final String ACTIVITY = "Activity ";
+
+    private final ActivityRepository activityRepository;
+    private final ActivityResourceAssembler activityResourceAssembler;
+    private final ProjectFiscalRepository projectFiscalRepository;
+    private final ProjectFiscalService projectFiscalService;
+    private final ActivityStatusCodeRepository activityStatusCodeRepository;
+    private final ContractPhaseCodeRepository contractPhaseCodeRepository;
+    private final RiskRatingCodeRepository riskRatingCodeRepository;
+
+    public ActivityService(
+            ActivityRepository activityRepository,
+            ActivityResourceAssembler activityResourceAssembler,
+            ProjectFiscalRepository projectFiscalRepository,
+            ProjectFiscalService projectFiscalService,
+            ActivityStatusCodeRepository activityStatusCodeRepository,
+            ContractPhaseCodeRepository contractPhaseCodeRepository,
+            RiskRatingCodeRepository riskRatingCodeRepository) {
+        this.activityRepository = activityRepository;
+        this.activityResourceAssembler = activityResourceAssembler;
+        this.projectFiscalRepository = projectFiscalRepository;
+        this.projectFiscalService = projectFiscalService;
+        this.activityStatusCodeRepository = activityStatusCodeRepository;
+        this.contractPhaseCodeRepository = contractPhaseCodeRepository;
+        this.riskRatingCodeRepository = riskRatingCodeRepository;
+    }
+
+    public CollectionModel<ActivityModel> getAllActivities(String projectGuid, String fiscalGuid) throws ServiceException {
+        try {
+            // Verify project fiscal exists and belongs to project
+            ProjectFiscalEntity projectFiscal = projectFiscalRepository.findById(UUID.fromString(fiscalGuid))
+                    .orElseThrow(() -> new EntityNotFoundException(FISCAL_NOT_FOUND + fiscalGuid));
```

Comment:
You could use MessageFormat.format("{0}: {1}", FISCAL_NOT_FOUND, fiscalGuid);
So you don't need to add the Colon and Space to the constant string. Same for other spots. Optional of course, what's here is fine.

Reply:
Good tip, I've updated now